classes/Cart.php

```diff
@@ -1754,15 +1758,17 @@ public function deleteProduct(
             );
         }
 
-        $preservedGifts = $this->getProductsGifts($id_product, $id_product_attribute);
-        if ($preservedGifts[(int) $id_product . '-' . (int) $id_product_attribute] > 0) {
-            return Db::getInstance()->execute(
-                'UPDATE `' . _DB_PREFIX_ . 'cart_product`
-                SET `quantity` = ' . (int) $preservedGifts[(int) $id_product . '-' . (int) $id_product_attribute] . '
-                WHERE `id_cart` = ' . (int) $this->id . '
-                AND `id_product` = ' . (int) $id_product .
-                ($id_product_attribute != null ? ' AND `id_product_attribute` = ' . (int) $id_product_attribute : '')
-            );
+        if ($preserveGiftsRemoval) {
+            $preservedGifts = $this->getProductsGifts($id_product, $id_product_attribute);
+            if ($preservedGifts[(int) $id_product . '-' . (int) $id_product_attribute] > 0) {
```

Comment:
```suggestion
            if (isset($preservedGifts[(int) $id_product . '-' . (int) $id_product_attribute]) && $preservedGifts[(int) $id_product . '-' . (int) $id_product_attribute] > 0) {
```

I think @PierreRambaud would appreciate that :)

Reply:
In this case, i will replace by a `!empty`